bin/common.bash

```diff
@@ -175,3 +179,51 @@ sops_exec_file_no_fifo() {
 
     sops --config "${sops_config}" exec-file --no-fifo "${1}" "${2}"
 }
+
+# Checks the current openstack env variables to see if anything is missing.
+# The user is then show what is set and prompted if they want to proceed or not.
+# The user can still proceed if nothing is set, to allow for other types of cloud providers.
+check_openstack_credentials() {
+    log_info "Checking for openstack user or openstack application credentials"
+
+    if [ -n "${OS_USERNAME:-}" ] && [ -n "${OS_APPLICATION_CREDENTIAL_NAME:-}" ]; then
+        log_warning "Warning: Both OS_USERNAME and OS_APPLICATION_CREDENTIAL_NAME is set."
+        log_warning "Will default to using Application credentials. You can unset application credentials to use openstack user instead."
+    fi
```

Comment:
Should I add this to be 100% sure what credentials are used?
```suggestion
    if [ -n "${OS_USERNAME:-}" ] && [ -n "${OS_APPLICATION_CREDENTIAL_NAME:-}" ]; then
        log_warning "Warning: Both OS_USERNAME and OS_APPLICATION_CREDENTIAL_NAME is set."
        log_warning "Will default to using Application credentials. You can unset application credentials to use openstack user instead."
        unset OS_USERNAME
        unset OS_PASSWORD
    fi
```

Reply:
here I would just exit and let the user decide what credentials he/she wants to use
maybe just print the unset commands for easy copy/paste

Reply:
Good idea. That might be the best solution.

Reply:
Fixed now.